File: C++/Solution995.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution995 {
public:
    int minKBitFlips(vector<int>& A, int K) {
        int count = 0, flip = 0;
        int size = A.size();
        vector<int> reverse(size);
        for(int i = 0; i < size; i++){
            flip = flip ^ reverse[i];
            if(flip == A[i]){
                if(i + K - 1 >= size){
                    return -1;
                }
                if(i + K - 1 < size - 1){
                    reverse[i + K] = reverse[i + K] ^ 1;
                }
                count++;
                flip = flip ^ 1;
            }
        }
        return count;
    }
};

File: Java/other/Solution995.java
class Solution995 {
    public int minKBitFlips(int[] A, int K) {
    	int size = A.length;
    	int count = 0;
    	for(int i = 0; i < size; i++){
    		if(A[i] == 0){
    			if(i + K - 1 >= size){
    				return -1;
    			}
    			for(int j = i; j <= i + K - 1; j++){
    				A[j] = A[j] ^ 1;
    			}
    			count++;
    		}
    	}
    	return count;
    }
}